frame/collective/src/lib.rs

```diff
@@ -121,165 +129,78 @@ impl DefaultVote for MoreThanMajorityThenPrimeDefaultVote {
-		/// a tally (yes votes and no votes given respectively as `MemberCount`).
-		/// \[account, proposal_hash, voted, yes, no\]
-		Voted(AccountId, Hash, bool, MemberCount, MemberCount),
-		/// A motion was approved by the required threshold.
-		/// \[proposal_hash\]
-		Approved(Hash),
-		/// A motion was not approved by the required threshold.
-		/// \[proposal_hash\]
-		Disapproved(Hash),
-		/// A motion was executed; result will be `Ok` if it returned without error.
-		/// \[proposal_hash, result\]
-		Executed(Hash, DispatchResult),
-		/// A single member did some action; result will be `Ok` if it returned without error.
-		/// \[proposal_hash, result\]
-		MemberExecuted(Hash, DispatchResult),
-		/// A proposal was closed because its threshold was reached or after its duration was up.
-		/// \[proposal_hash, yes, no\]
-		Closed(Hash, MemberCount, MemberCount),
-	}
-}
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T, I=()>(PhantomData<(T, I)>);
 
-decl_error! {
-	pub enum Error for Module<T: Config<I>, I: Instance> {
-		/// Account is not a member
-		NotMember,
-		/// Duplicate proposals not allowed
-		DuplicateProposal,
-		/// Proposal must exist
-		ProposalMissing,
-		/// Mismatched index
-		WrongIndex,
-		/// Duplicate vote ignored
-		DuplicateVote,
-		/// Members are already initialized!
-		AlreadyInitialized,
-		/// The close call was made too early, before the end of the voting.
-		TooEarly,
-		/// There can only be a maximum of `MaxProposals` active proposals.
-		TooManyProposals,
-		/// The given weight bound for the proposal was too low.
-		WrongProposalWeight,
-		/// The given length bound for the proposal was too low.
-		WrongProposalLength,
-	}
-}
-
-/// Return the weight of a dispatch call result as an `Option`.
-///
-/// Will return the weight regardless of what the state of the result is.
-fn get_result_weight(result: DispatchResultWithPostInfo) -> Option<Weight> {
-	match result {
-		Ok(post_info) => post_info.actual_weight,
-		Err(err) => err.post_info.actual_weight,
-	}
-}
+	/// Origin for the collective module.
+	#[pallet::origin]
+	pub type Origin<T, I = ()> = RawOrigin<<T as frame_system::Config>::AccountId, I>;
 
+	#[pallet::hooks]
+	impl<T: Config<I>, I: 'static> Hooks<BlockNumberFor<T>> for Pallet<T, I> {
+		fn on_runtime_upgrade() -> frame_support::weights::Weight {
+			if !UpgradedToTripleRefCount::<T, I>::get() {
+				UpgradedToTripleRefCount::<T, I>::put(true);
+				migrations::migrate_to_triple_ref_count::<T, I>()
+			} else {
+				0
+			}
+		}
 
-// Note that councillor operations are assigned to the operational class.
-decl_module! {
-	pub struct Module<T: Config<I>, I: Instance=DefaultInstance> for enum Call where origin: <T as frame_system::Config>::Origin {
-		type Error = Error<T, I>;
+		fn integrity_test() {
+			T::BlockWeights::get()
+				.validate()
+				.expect("The weights are invalid.");
+		}
```

Comment:
this is new ? why do we do this now ?

Reply:
Sure will remove it, Sorry its my bad, took it from `frame/system/src/lib.rs`,
assumed that it's part of trait Hooks<> pattern.